File: AS-2023-2024/5X-INFO/es-11_2024-01-16_PeopleCRUD/v2.0-spring_jpa/src/main/java/pkg/router/PersonRouter.java
package pkg.router;
import java.util.List;
import pkg.entity.Person;
import pkg.repository.PersonRepository;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
@RestController
@RequestMapping("/api/person")
public class PersonRouter {
	@Autowired
	private PersonRepository people;
	@PostMapping			// ./ JSON Person in the body
	public ResponseEntity<String> createPerson(@RequestBody Person person) {
		if (!people.existsById(person.getId())) {
			people.save(person);
			return new ResponseEntity<>(
				"Success! Person '%s' has been added to the list!".formatted(person.toString()),
				HttpStatus.OK
			);
		}
		return new ResponseEntity<>(
			"Unable to add Person '%s'! Same ID already exists!".formatted(person.toString()),
			HttpStatus.BAD_REQUEST
		);
	}
	@GetMapping				// ./
	public List<Person> getPeople() {
		return people.findAll();
	}
	@GetMapping("/{id}")	// ./5
	public ResponseEntity<Person> getPerson(@PathVariable int id) {
		return people.findById(id)
			.map(p -> new ResponseEntity<>(p, HttpStatus.OK))
			.orElse(new ResponseEntity<>(HttpStatus.NOT_FOUND));
	}
	@GetMapping("/")		// ./?id=7
	public ResponseEntity<Person> getPersonByQueryURL(@RequestParam int id) {
		return getPerson(id);
	}
	@PutMapping("/{id}")	// ./11
	public ResponseEntity<String> modifyPerson(@PathVariable int id, @RequestBody Person person) {
		if (people.existsById(id)) {
			person.setId(id);
			people.save(person);
			return new ResponseEntity<>(
				"Person with ID %d updated successfully".formatted(id),
				HttpStatus.OK
			);
		}
		return new ResponseEntity<>(
			"Unable to update Person with ID %d. Person not found.".formatted(id),
			HttpStatus.NOT_FOUND
		);
	}
	@DeleteMapping("/{id}")	// ./3
	public ResponseEntity<String> removePerson(@PathVariable int id) {
		if (people.existsById(id)) {
			people.deleteById(id);
			return new ResponseEntity<>("Success! Correct remove Person ID %d!".formatted(id), HttpStatus.OK);
		}
		return new ResponseEntity<>("Error! Unable remove Person ID %d".formatted(id), HttpStatus.NOT_FOUND);
	}
}

File: AS-2023-2024/5X-TPSIT/es-09_2024-01-15_PeopleCRUD/v1.8-sql/routers/PersonRouter.py
from entity.Person import Person
from fastapi import APIRouter, HTTPException
from db.database import selectFromTab, updateFromTab

router = APIRouter(
	tags = ['Person'],
	prefix = '/api/person'
)

@router.post('/')
async def createPerson(p: Person):
	if updateFromTab(f'INSERT INTO People(age, lastname, firstname) VALUES({p.age}, "{p.lastname}", "{p.firstname}");') > 0:
		raise HTTPException(detail = 'Unable to add new Person! Maybe same ID already exists!', status_code = 404)
	return 'Success! A new person has been added to the list!'

@router.get('/')
async def getPeople():
	return selectFromTab()

@router.get('/{id}')
async def getPerson(id: int):
	p = selectFromTab(f'id={id}')
	if len(p) > 0:
		return p[0]
	raise HTTPException(detail = f'ID Person {id} not found!..', status_code = 404)

@router.put('/{id}')
async def modifyPerson(id: int, p: Person):
	if updateFromTab(f"UPDATE People SET age={p.age}, lastname='{p.lastname}', firstname='{p.firstname}' WHERE id={id};") > 0:
		return f'Person with ID {id} updated successfully'
	raise HTTPException(detail = f'Error! Unable remove Person id {id}', status_code = 404)

@router.delete('/{id}')
async def removePerson(id: int):
	if updateFromTab(f'DELETE From People WHERE id={id};') > 0:
		return f'Success! Correct remove Person ID {id}!'
	raise HTTPException(detail = f'Error! Unable remove Person id {id}', status_code = 404)